### ISSUE TITLE
grafana_team_external_group import issue

### ISSUE THREAD

**Issue by marijus-ravickas** [grafana/terraform-provider-grafana#796]
### Terraform Version
* Terraform: v1.3.7
* Terraform Grafana Provider: 1.34
* Grafana: v9.2.8.2

### Affected Resource(s)
Please list the resources as a list, for example:
* grafana_team_external_group

### Issue reproducing. Only related to terraform import

```Terraform
resource "grafana_team" "accounting" {
  name = "Accounting"
}

resource "grafana_team_external_group" "accounting-group" {
  team_id = grafana_team.accounting.team_id
  groups  = ["Accounting", "infra-grafana-${var.grafana_environment}-bu-accounting"]
}
```

Example above works, just during import there is a bug. More details. No need to import `grafana_team` resource, but this looks more logical, so let's do it together to not loose couples.  

```shell
terraform import -var-file=main.tfvars grafana_team.accounting 28 
```
Next let's get `grafana_team_external_group` in place. 

```shell
terraform import -var-file=main.tfvars grafana_team_external_group.accounting-group 28 
```

Import should be successful, but if any issues, usualy this is related to typos or any non technical thing. Focus. :)

Next step would be to verify if plan and apply is going fine. And `terraform plan` looks ok, just `grafana_team_external_group.accounting-group` is marked for replacement, but oh well, not acritical thing as it will be added back. But after `terraform apply` during resource replacement we are getting http error 400, as group is already exist. 

```
Terraform will perform the following actions:

  # grafana_team_external_group.accounting-group must be replaced
-/+ resource "grafana_team_external_group" "accounting-group" {
      ~ id      = "28" -> (known after apply)
      + team_id = 28 # forces replacement
        # (1 unchanged attribute hidden)
    }

Plan: 1 to add, 0 to change, 1 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

grafana_team_external_group.accounting-group: Destroying... [id=28]
grafana_team_external_group.accounting-group: Destruction complete after 0s
grafana_team_external_group.accounting-group: Creating...
╷
│ Error: status: 400, body: {"message":"Group is already added to this team"}
│
│   with grafana_team_external_group.accounting-group,
│   on team.tf line 5, in resource "grafana_team_external_group" "accounting-group":
│    5: resource "grafana_team_external_group" "accounting-group" {
│
╵
```

After some try/error, noticed in state that during import `"team_id": null` which causes new group creation during `terraform apply`.

```
    {
      "mode": "managed",
      "type": "grafana_team_external_group",
      "name": "accounting-group",
      "provider": "provider[\"registry.terraform.io/grafana/grafana\"]",
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "groups": [
              "Accounting",
              "infra-grafana-staging-bu-accounting"
            ],
            "id": "28",
            "team_id": null
          },
          "sensitive_attributes": [],
          "private": "eyJzY2hlbWFfdmVyc2lvbiI6IjAifQ=="
        }
      ]
    },
```

So, dirty fix, not recommended. Replace `"team_id"` value with `int('id')` from instance attribute. After this everything works as expected, state is updated and terraform is now managing this resource.

**Comment by marijus-ravickas** [comment]
Looks like recreation is caused by `ForceNew:    true,` in resource schema, this might require some additional logic. 
https://github.com/grafana/terraform-provider-grafana/blob/master/grafana/resource_team_external_group.go#L31

**Comment by eleijonmarck** [comment]
I *can* reproduce.

The steps I have taken is:

1. create a team within grafana
2. create a group in the external-group-sync for that team
3. importing both yields no errors for me.
4. terraform apply (which gives me the error)

```bash
1 % terraform import grafana_team.external3 3
grafana_team.external3: Importing from ID "3"...
grafana_team.external3: Import prepared!
  Prepared grafana_team for import
grafana_team.external3: Refreshing state... [id=3]

Import successful!

The resources that were imported are shown above. These resources are now in
your Terraform state and will henceforth be managed by Terraform.
```

importing the externalgroup

```bash
% terraform import grafana_team_external_group.external3-group 3
grafana_team_external_group.external3-group: Importing from ID "3"...
grafana_team_external_group.external3-group: Import prepared!
  Prepared grafana_team_external_group for import
grafana_team_external_group.external3-group: Refreshing state... [id=3]

Import successful!

The resources that were imported are shown above. These resources are now in
your Terraform state and will henceforth be managed by Terraform.
```

this is my terraform configuration:
```tf
terraform {
  required_providers {
    grafana = {
      source = "grafana/grafana"
    }
  }
}

// Configure the Grafana Provider
provider "grafana" {
  url  = "http://localhost:3000/"
  auth = "admin:admin"
}

resource "grafana_team" "external3" {
  name = "external3"
}

resource "grafana_team_external_group" "external3-group" {
  team_id = grafana_team.external3.team_id
  groups  = ["external3", "infra-grafana-group-bu-accounting"]
}
```

might be that I am using the latest provider and grafana. But no changes have really been made on this since the versions that the user provided

```bash
% terraform apply
grafana_team.external3: Refreshing state... [id=3]
grafana_team.accounting: Refreshing state... [id=1]
grafana_team_external_group.accounting-group: Refreshing state... [id=2]
grafana_team_external_group.external3-group: Refreshing state... [id=3]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place
-/+ destroy and then create replacement

Terraform will perform the following actions:

  # grafana_team.accounting will be updated in-place
  ~ resource "grafana_team" "accounting" {
        id                               = "1"
      + ignore_externally_synced_members = true
      ~ members                          = [
          - "ldap-editor",
        ]
        name                             = "test"
        # (1 unchanged attribute hidden)
    }

  # grafana_team.external3 will be updated in-place
  ~ resource "grafana_team" "external3" {
        id                               = "3"
      + ignore_externally_synced_members = true
        name                             = "external3"
        # (2 unchanged attributes hidden)
    }

  # grafana_team_external_group.accounting-group must be replaced
-/+ resource "grafana_team_external_group" "accounting-group" {
      ~ groups  = [
          + "Accounting",
          - "external-group",
          + "infra-grafana-group-bu-accounting",
        ]
      ~ id      = "2" -> (known after apply)
      + team_id = 1 # forces replacement
    }

  # grafana_team_external_group.external3-group must be replaced
-/+ resource "grafana_team_external_group" "external3-group" {
      ~ groups  = [
          + "infra-grafana-group-bu-accounting",
            # (1 unchanged element hidden)
        ]
      ~ id      = "3" -> (known after apply)
      + team_id = 3 # forces replacement
    }

Plan: 2 to add, 2 to change, 2 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

grafana_team_external_group.accounting-group: Destroying... [id=2]
grafana_team_external_group.external3-group: Destroying... [id=3]
grafana_team_external_group.external3-group: Destruction complete after 0s
grafana_team_external_group.accounting-group: Destruction complete after 0s
grafana_team.accounting: Modifying... [id=1]
grafana_team.external3: Modifying... [id=3]
grafana_team.external3: Modifications complete after 0s [id=3]
grafana_team_external_group.external3-group: Creating...
grafana_team.accounting: Modifications complete after 0s [id=1]
grafana_team_external_group.accounting-group: Creating...
grafana_team_external_group.accounting-group: Creation complete after 0s [id=1]
╷
│ Error: status: 400, body: {"message":"Group is already added to this team"}
│
│   with grafana_team_external_group.external3-group,
│   on main.tf line 28, in resource "grafana_team_external_group" "external3-group":
│   28: resource "grafana_team_external_group" "external3-group" {
│
╵
```

**Comment by marijus-ravickas** [comment]
You did great, but just half of the reproduction. After importing try to apply state.

**Comment by eleijonmarck** [comment]
ah i read more of the description now :D , il adjust my comment accordingly

**Comment by eleijonmarck** [comment]
fyi; for anyone working (mostly for myself). Running apply only to create these resources does not reproduce the error (running apply twice). It is only on import that the apply does not have the correct state

**Comment by eleijonmarck** [comment]
We have a PR up for review for this fix

**Comment by eleijonmarck** [comment]
Essentially, the Reader of the external_group resource was not taking the `team_id` from the state when reading. This should fix the issue

**Comment by julienduchesne** [comment]
Fixed in https://github.com/grafana/terraform-provider-grafana/pull/846. Released in v1.36.0
